### PR TITLE
Features -> Enhancements

### PR DESCRIPTION
**Pull request by nikopen**
@justaugustus @kacole2 

Needs more changes but probably not all `features` should be replaced, PTAL

### CODE REVIEW

File: contributors/devel/release.md [hunk 1]
```diff
@@ -141,11 +141,11 @@ project is always stable so that individual commits can be
...
 
 With ongoing feature definition through the year, some set of items
-will bubble up as targeting a given release.  The **feature freeze**
+will bubble up as targeting a given release.  The **enhancement freeze**
 starts ~4 weeks into release cycle.  By this point all intended
 feature work for the given release has been defined in suitable
-planning artifacts in conjunction with the Release Team's [features
-lead](https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/features).
+planning artifacts in conjunction with the Release Team's [enhancements
+lead](https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/enhancements).
```
Comment (justaugustus) [inline contributors/devel/release.md]: Replace w/ `https://git.k8s.io/sig-release/release-team/role-handbooks/enhancements/README.md`

**Comment by justaugustus** [comment]
One nit. LGTM otherwise. Thanks for the cleanup, @nikopen!

**Comment by nikopen** [comment]
done!

**Comment by justaugustus** [comment]
@nikopen -- one more nit. Can you squash the commits? :)

**Comment by cblecker** [comment]
/approve
LGTM to @justaugustus

**Pull request merged**